### PR TITLE
Collect metrics with Prometheus C++ Client

### PR DESCRIPTION
**Pull request by gaschler**
RFC=[0014](https://github.com/googlecartographer/rfcs/blob/master/text/0014-monitoring.md)

### CODE REVIEW

File: cartographer_grpc/metrics/prometheus/metrics_test.cc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  ASSERT_EQ(collected[0].metric_size(), 1);
+  EXPECT_THAT(
+      collected[0].metric(0).label(),
+      testing::AllOf(
+          testing::ElementsAre(testing::Property(
+              &io::prometheus::client::LabelPair::name, kLabelKey)),
+          testing::ElementsAre(testing::Property(
+              &io::prometheus::client::LabelPair::value, kLabelValue))));
+  EXPECT_THAT(collected[0].metric(0).counter().value(),
+              testing::Eq(expected_value));
```
Comment (stevewolter) [inline cartographer_grpc/metrics/prometheus/metrics_test.cc]: Please make sure that testing::Eq is safe for comparing doubles (has an epsilon for equality), or tests will become flaky.

Comment (gaschler) [reply cartographer_grpc/metrics/prometheus/metrics_test.cc]: Done. Thanks for the fast review.

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Pull request merged**